client/checkout/woopay/direct-checkout/woopay-direct-checkout.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Internal dependencies
+ */
+import { getConfig } from 'wcpay/utils/checkout';
+import ReactDOM from 'react-dom';
+import { WooPayConnectIframe } from 'wcpay/checkout/woopay/direct-checkout/woopay-connect-iframe';
+import request from 'wcpay/checkout/utils/request';
+import { buildAjaxURL } from 'wcpay/payment-request/utils';
+
+class WoopayDirectCheckout {
+	static state = {
+		iframePostMessage: null,
+	};
+	static listeners = {
+		setIframePostMessage: () => {},
+		setTempThirdPartyCookie: () => {},
+		setIsWooPayThirdPartyCookiesEnabled: () => {},
+		setIsUserLoggedIn: () => {},
+		setWoopaySessionResponse: () => {},
+	};
+	static actionCallback = {
```

Comment:
Based on our discussion it might be helpful to drop a brief description of this functionality in the PR description for future reference.

Reply:
Good call. I ended up adding inline comments for easy reference in 446f6ee7a850c51d7e975e2c164b46409959b724.